api/models.py

```diff
@@ -240,23 +240,9 @@ class Node(DatabaseModel):
     def update(self):
         self.updated = datetime.utcnow()
 
-    @classmethod
-    def validate_state(cls, state):
-        """Validate Node.state"""
-        if state and state not in [state.value for state in StateValues]:
-            raise ValueError(f"Invalid state value '{state}'")
-
-    @classmethod
-    def validate_result(cls, result):
-        """Validate Node.result"""
-        if result and result not in [result.value for result in ResultValues]:
-            raise ValueError(f"Invalid result value '{result}'")
-
```

Comment:
These methods are validating GET request query parameters for `/nodes`. It is not for the validation of POST node requests.

```
curl http://localhost:8001/latest/nodes?state=don
{"detail":"Invalid state value 'don'"}
```
We won't get any intimation of an invalid value without the validation:
```
$ curl http://localhost:8001/latest/nodes?state=don
{"items":[],"total":0,"limit":50,"offset":0}
```

Reply:
I guess it is fine to drop these validations as it will work as expected if the value does not match with provided `state` or `result`.

Reply:
Ah OK, so I might need to tweak the commit message.

Reply:
Also there's nothing in the model class that shows it's only used for GET queries.  Then I wonder if we also need the `PyObjectId` validation.  Well I made this change initially as a side-effect of a bug I found with the default value of `Node.state` which had the `Enum` object instance rather than a plain string.  I'll go through this validation code again to double check.